src/main/java/com/github/complate/nashorn/renderer/NashornComplateRenderer.java

```diff
@@ -37,10 +45,29 @@
             "    var console = { log: print, error: print };\n" +
             "}\n\n";
 
-    private final NashornScriptEngine engine;
+    private final ThreadLocal<NashornScriptEngine> engine = new ThreadLocal<NashornScriptEngine>() {
```

Comment:
This could be a place for using a "lambda" method instead of an anonymous instance. What is your take on using these here?